src/frequently-requested-changes.md

```diff
@@ -0,0 +1,165 @@
+design philosophy. For instance, we don't want to make changes that make code
+easier to write but harder to read, or changes that make code more error-prone
+to modify and maintain.
+
+That said, *we are open to proposals that involve new syntax*, especially for
+new features, or to improve an existing fundamental feature. The bar for new
+syntax (e.g. new operators) is high, but not insurmountable. But the bar for
+*changes to existing syntax* is even higher.
+
+## Changes to avoid writing `self.method()` when calling a method from a method
+
+In Rust, within a method of an object, calling another method requires writing
+`self.othermethod()`, and accessing a field requires `self.field`. In some
+other languages, such accesses can omit the equivalent of `self.`, and just
+write `othermethod` or `field`, implicitly referencing the "current object".
+People writing code with such method calls or field accesses sometimes ask for
+such shorthand in Rust.
+
+Rust prefers the explicitness of writing `self.` for method calls and field
+accesses, and in most ways treating `self` as a normal object of the type. This
+avoids unexpected calls to the wrong method, makes it easier to distinguish
+methods and free functions, and makes code easier to read. We don't want to
+provide a shorthand that makes this syntax briefer at the expense of this
+clarity and unambiguity.
+
+## Arbitrary custom operator syntax
+
+Rust allows overloading existing operators; for instance, you can implement the
+`Add` trait to overload the `+` operator. However, Rust does not allow creating
+*new* operators.
+
+Some would argue this can make code more readable, if you know what the
+operators mean. If you don't, it makes the code inscrutable.
+
+In general, such a change would substantially raise barriers to entry for Rust
+developers, making Rust code less approachable and less universally
+understandable. We're unlikely to add support for this.
+
+Note that Rust's existing operator overloading is semantic (`impl Add`), not
+syntactic (`impl Plus`), which tends to encourage using overloaded operators
+for the same semantic purposes, rather than for building arbitrary domain-specific languages.
+
+Rust developers seeking to build arbitrary domain-specific languages (DSLs)
+should consider the macro system.
+
+## Numeric overflow checking should be on by default even in release mode
+
+Whenever possible, Rust tries to do the safe thing by default.
+
+Numeric overflow checking (e.g. `1000u16 * 1000u16`) is one case where Rust
+compromised on this: on many targets, numeric overflow checking has high enough
+overhead to hurt performance too much for a wide variety of code. As a result,
+Rust defaults to having overflow checking only for debug builds, while release
+builds have overflow checking off by default. (In release builds, numeric
+overflow wraps, but code cannot count on this behavior, and projects can turn
```

Comment:
I think the phrase "code cannot count on this behavior" is overly scary for the casual reader.

At least, I believe our intention has always been that for an operation `A op B`, *either* you will get a panic (unwind or abort) or you will see `A op B mod 2^bitwidth` as a result, and that to be a guarantee that one can rely on in one's library code. Is your intent here to *not* provide that guarantee?

Reply:
My intent was "you can't count on non-panicking", nothing more than that. I'm happy to clarify that.